WordPress/Classes/Services/JetpackSocialService.swift

```diff
@@ -19,8 +19,11 @@ import WordPressKit
 
     /// Init method for Objective-C.
     ///
-    @objc init(contextManager: ContextManager) {
-        self.coreDataStack = contextManager
+    @objc init(contextManager: CoreDataStack) {
+        guard let typeCastStack = contextManager as? CoreDataStackSwift else {
+            fatalError("Expected a CoreDataStackSwift-conforming type even though this initializer is marked @objc.")
+        }
+        self.coreDataStack = typeCastStack
```

Comment:
Not proud of this one. I don't know if there's a neater way to go about it because if `CoreDataStackSwift` could be defined in Objective-C, then we wouldn't have needed it in the first place.

Luckily, this hack has to be used only twice, so I think we can live with it...?

Of course, one possible solution would be to convert the Objective-C consumers to Swift... But I think we ought to focus on moving forward with the Core Data migration at the moment.

Reply:
IMO, it'd be okay to remove the argument and use the shared instance internally.